build/ci_utils/src/cache/goodie/graalvm.rs

```diff
@@ -54,27 +52,40 @@ impl Goodie for GraalVM {
 
     fn is_active(&self) -> BoxFuture<'static, Result<bool>> {
         let expected_graal_version = self.graal_version.clone();
-        let expected_java_language_version = self.java_version;
         async move {
             let found_version = find_graal_version().await?;
             ensure!(found_version == expected_graal_version, "GraalVM version mismatch. Expected {expected_graal_version}, found {found_version}.");
-
-            let found_java_version = Java.check_language_version().await?;
-            ensure!(
-                found_java_version == expected_java_language_version,
-                "Java language version mismatch. Expected {expected_java_language_version}, found {found_java_version}."
-            );
-
             Result::Ok(true)
         }
         .boxed()
     }
 
     fn activation_env_changes(&self, package_path: &Path) -> Result<Vec<crate::env::Modification>> {
-        let package_path = package_path.join(self.root_directory_name());
+        let dir_entries = package_path
+            .read_dir()
+            .context("Failed to read GraalVM cache directory")
+            .unwrap()
+            .collect_vec();
+        assert_eq!(
+            dir_entries.len(),
+            1,
+            "GraalVM cache directory should contain exactly one directory"
+        );
+        let graalvm_dir = match dir_entries.get(0).unwrap() {
+            Ok(dir_entry) => dir_entry,
+            Err(err) => bail!("Failed to read GraalVM cache directory: {}", err),
+        };
+        let dir_name_tmp = graalvm_dir.file_name();
```

Comment:
It's not really a temp, if (as I suppose) you added this to make a result of temporary expression non-temporary.
I'd just name this `dir_name` and shadow the name next. The value will live on, unaffected.

Reply:
Yes, that was exactly the purpose. Fixed in 2064a5ca162fd6ba02b4e4c7f1d354b403732f8b by shadowing the `dir_name` variable.